tests/baselines/reference/ambientShorthand_reExport.js

```diff
@@ -24,7 +24,18 @@ exports.x = jquery_1.x;
 //// [reExportAll.js]
 "use strict";
 function __export(m) {
-    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
+    for (var p in m) b(p);
+    function b(p) {
+        if (!exports.hasOwnProperty(p))
+            Object.create

```

Comment:
Any chance you can update this for this feedback? https://github.com/microsoft/TypeScript/pull/33587#discussion_r362659106

> Not a fan of how this gets checked on every iteration of the loop. In other places, we just decide up front what the helper is going to be and use that.

Reply:
We _could_, but it'd double the helper size. In the case of `__importStar`, we'd go from
```ts
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) b(k);
    result["default"] = mod;
    return result;
    function b(p) {
        if (Object.hasOwnProperty.call(mod, p))
            Object.create
                ? Object.defineProperty(result, p, {
                      enumerable: true,
                      get: function() {
                          return mod[p];
                      }
                  })
                : (result[p] = mod[p]);
    }
};
```

to
```js
var __importStar = (this && this.__importStar) || (Object.create ? (function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) b(k);
    result["default"] = mod;
    return result;
    function b(p) {
        if (Object.hasOwnProperty.call(mod, p))
            Object.defineProperty(result, p, {
                enumerable: true,
                get: function() {
                    return mod[p];
                }
            });
    }
}) : (function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) b(k);
    result["default"] = mod;
    return result;
    function b(p) {
        if (Object.hasOwnProperty.call(mod, p))
             result[p] = mod[p];
    }
}));
```
(you can compress it a bit more with another helper to join the shared bits between the two implementations, but not much, and that's much more complexity)

I think doing the (hopefully fast) check on the (executed only a few times at the top level) import/export helpers is a fair enough tradeoff to make in our default helper implementations. Really: If you have strong execution speed or size concerns, you'll be providing custom helpers that target exactly the runtime you're building for, without any runtime feature testing at all.

Reply:
> (you can compress it a bit by moving the property definition into another helper (eg `__setBinding`) but that's way more complexity, and if we wanted to do that, I'd rather do it in a followup)

Yeah, that's what I had in mind. But I guess we can improve in another PR if users ask.

Reply:
You also could have done this:
```js
var __importStar = (this && this.__importStar) || (function () {
  var setModuleDefault = (Object.create ? (function(o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function(o, v) {
        o["default"] = v;
    });
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
        setModuleDefault(result, mod);
        return result;
    };
})();

Reply:
Or this:
```js
var __importStar = (this && this.__importStar) || (function (setModuleDefault) {
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
        setModuleDefault(result, mod);
        return result;
    };
})(Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
```

Reply:
See, it's used by both `__importStar` _and_ `__exportStar`, though, so it actually saves space by being a separate helper.

Reply:
We already do that for `__extends` in tslib. In `src/compiler/transformers/es2015.ts` we have `extendsStatics` inline inside of `__extends`, but in tslib they are two distinct functions.

Reply:
We also defer its value until later in case a polyfill adds `Object.setPrototypeOf`, but you can't properly polyfill `Object.defineProperty` due to IE8.